### PR TITLE
Added option to Flink sink to throw an exception on dropped messages.

### PR DESCRIPTION
**Pull request by gesundkrank**
Also updated Flink dep to latest version.

### CODE REVIEW

File: flink/src/main/scala/com/metamx/tranquility/flink/BeamSink.scala [hunk 1]
```diff
@@ -32,8 +32,9 @@ import org.apache.flink.streaming.api.functions.sink.RichSinkFunction
   * This class provides a sink that can propagate any event type to Druid.
   *
   * @param beamFactory your implementation of [[BeamFactory]].
+  * @param throwOnDrop throws an exception if set to true and a message gets dropped.
   */
-class BeamSink[T](beamFactory: BeamFactory[T])
+class BeamSink[T](beamFactory: BeamFactory[T], throwOnDrop: Boolean = false)
```
Comment (gianm) [inline flink/src/main/scala/com/metamx/tranquility/flink/BeamSink.scala]: This is called `reportDropsAsExceptions` in other modules, could you use that name here?

Comment (gesundkrank) [inline flink/src/main/scala/com/metamx/tranquility/flink/BeamSink.scala]: Sure. Committed a fixup.

**Comment by gianm** [comment]
👍  thanks!

btw, do you have any thoughts on #161?

**Comment by gesundkrank** [comment]
Thank you! 

Haven't seen the issue. Thanks for the hint. Will open a PR that fixes it, soon.

**Comment by gianm** [comment]
@gesundkrank sweet, thanks!

**Pull request merged**